frontend/js/tests/user/stats/UserArtistMap.test.tsx

```diff
@@ -1,247 +1,145 @@
-
-      const spy = jest.spyOn(instance.APIService, "getUserArtistMap");
-      spy.mockImplementation(() =>
-        Promise.resolve(userArtistMapResponse as UserArtistMapResponse)
-      );
-      const result = await instance.getData();
-
-      expect(spy).toHaveBeenCalledWith(props?.user?.name, "week");
-      expect(result).toEqual(userArtistMapResponse);
-    });
-
-    it("sets state correctly if data is not calculated", async () => {
-      const wrapper = shallow<UserArtistMap>(<UserArtistMap {...props} />);
-      const instance = wrapper.instance();
-
-      const spy = jest.spyOn(instance.APIService, "getUserArtistMap");
-      const noContentError = new APIError("NO CONTENT");
-      noContentError.response = {
-        status: 204,
-      } as Response;
-      spy.mockImplementation(() => Promise.reject(noContentError));
-      await act(async () => {
-        await instance.getData();
-      });
-      await waitForComponentToPaint(wrapper);
-
-      expect(wrapper.state()).toMatchObject({
-        loading: false,
-        hasError: true,
-        errorMessage: "NO CONTENT",
-      });
-    });
+  afterAll(() => {
+    server.close();
   });
 
-  describe("processData", () => {
-    it("processes data correctly for all_time", () => {
-      const wrapper = shallow<UserArtistMap>(
-        <UserArtistMap {...{ ...props, range: "all_time" }} />
-      );
-      const instance = wrapper.instance();
-
-      const result = instance.processData(
-        userArtistMapResponse as UserArtistMapResponse,
-        "artist"
-      );
-
-      expect(result).toEqual(userArtistMapProcessedDataArtist);
+  it("renders correctly", async () => {
+    renderWithProviders(
+      <UserArtistMap {...props} />,
+      {},
+      {
+        wrapper: reactQueryWrapper,
+      }
+    );
+
+    // Wait for loading to finish
+    await waitFor(() => {
+      expect(screen.getByTestId("Choropleth")).toBeInTheDocument();
     });
 
-    it("processes data correctly for listen", () => {
-      const wrapper = shallow<UserArtistMap>(
-        <UserArtistMap {...{ ...props, range: "all_time" }} />
-      );
-      const instance = wrapper.instance();
-
-      const result = instance.processData(
-        userArtistMapResponse as UserArtistMapResponse,
-        "listen"
-      );
-
-      expect(result).toEqual(userArtistMapProcessedDataListen);
-    });
-    it("returns an empty array if no payload", () => {
-      const wrapper = shallow<UserArtistMap>(
-        <UserArtistMap {...{ ...props, range: "all_time" }} />
-      );
-      const instance = wrapper.instance();
-
-      // When stats haven't been calculated, processData is called with an empty object
-      const result = instance.processData(
-        {} as UserArtistMapResponse,
-        "listen"
-      );
-
-      expect(result).toEqual([]);
-    });
+    expect(screen.getByTestId("user-stats-map")).toBeInTheDocument();
+    expect(screen.getByTestId("Choropleth")).toBeInTheDocument();
```

Comment:
I think you're testing the same thing on line 100, can probably remove this line.